game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameSelectorModel.java

```diff
@@ -186,10 +189,28 @@ public void loadDefaultGameSameThread() {
         // can load. (ie: if a previous version of triplea was using running a game within its
         // root folder, we shouldn't open it)
         .filter(
-            defaultGame ->
-                defaultGame.contains(ClientFileSystemHelper.getUserRootFolder().toURI().toString()))
-        .filter(defaultGame -> new File(defaultGame).exists())
+            defaultGame -> {
+              String rootFolder = ClientFileSystemHelper.getUserRootFolder().toURI().toString();
+              if (rootFolder.startsWith("file:")) {
+                // strip off this prefix so it matches better against the defaultGame string
+                rootFolder = rootFolder.substring("file:".length());
+              }
+              return defaultGame.contains(rootFolder);
+            })
+        // ensure the default game hasn't been deleted since it was last loaded
+        .filter(
+            defaultGame -> {
+              if (defaultGame.startsWith("jar:file:")) {
+                // when the game is from a zip file, it starts with jar:file: and uses a '!'
+                // to separate the local file from the zip content
+                defaultGame = defaultGame.substring("jar:file:".length(), defaultGame.indexOf('!'));
+              } else if (defaultGame.startsWith("file://")) {
+                // when the game is from a local file, it starts with file://
+                defaultGame = defaultGame.substring("file://".length());
+              }
```

Comment:
The lack of an else implies that the default game can be a third case, that would be valid. If not valid, then we keep processing rather than flagging an error here, which I think the "fail early" principle would suggest that it would be better to throw an exception or some other error handling. Even better probably would be to ensure that we cannot have an invalid value set in default game, and if so to remove it, all before we get to this point. WDYT?

Reply:
Second thought, can we use any of the 'isFile' APIs or otherwise to check if we are dealing with a zip file?


This SO seems to have one suggestion we could use:

```
String contentType = Files.probeContentType(Paths.get(pathToFile);

if (!contentType.equals("application/zip")) {

    throw new NotZipFileException();
}
```

https://stackoverflow.com/questions/33934178/how-to-identify-a-zip-file-in-java

I suspect this is another candidate method to be added to `FileUtils` so we can remove this logic from `GameSelectorModel`. The game selector model certainly has a number of issues, but it knowing how to do file system processing and detection seems to be adding to SRP violation, so I am very much in favor of removing complexity from it to a helper to simplify the file system ops.

Reply:
I got `Path.of` to work but it wasn't as easy as just calling `Path.of`.  If the uri is "jar:file:/location/to/file.zip!/location/to/map.xml", then `Path.of` will return a path pointing to the xml file.  But that file is inside of the zip file and isn't what we want.  But the first file store of the Path object is the "zipfs" store which has a name of the actual zip file.  And that can be used to create a Path.

Once I did that, the code simplified to just using Path objects.  I believe this should be cross-platform.